### PR TITLE
Set a default value for diff_package_changed

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by evgeni** [comment]
Shouldn't this be covered by

https://github.com/ehelms/obal/blob/76e1b2aa8ec64c7b4140ca96edb494ad074706c4/obal/data/roles/diff_package/tasks/main.yml#L8-L11

**Comment by ekohl** [comment]
Could we disable coveralls output messages like those?

**Comment by ehelms** [comment]
Nope. I can show you the error -- http://ci.theforeman.org/job/tfm-ror51-release/75/console

**Comment by evgeni** [comment]
So I think the issue is that none of the tasks in https://github.com/ehelms/obal/blob/76e1b2aa8ec64c7b4140ca96edb494ad074706c4/obal/data/roles/diff_package/tasks/copr.yml are tagged with `always`, and not that we don't define a default

**Comment by ehelms** [comment]
Defining a default fixed the problem locally. So are we talking about two different issues? Or the same issue just two ways to skin the cat?

**Comment by evgeni** [comment]
did you run it locally with `--tags wait`? the problem (IMHO) is that due to `--tags wait`, all tasks in https://github.com/ehelms/obal/blob/76e1b2aa8ec64c7b4140ca96edb494ad074706c4/obal/data/roles/diff_package/tasks/copr.yml are skipped, and then the variable is really not defined.

so yes, defining a default of true "fixes" it, by means that it will rebuild copr always :)

(that said, I think we should err on the side of defaulting to true, but `copr.yaml` still needs fixes)

File: obal/data/roles/diff_package/tasks/copr.yml [hunk 1]
```diff
@@ -10,24 +10,36 @@
       --with-latest-build
   register: copr_package_info
   ignore_errors: true
+  tags:
```
Comment (evgeni) [inline obal/data/roles/diff_package/tasks/copr.yml]: `include_tasks: git_package_info.yml` also needs tags

**Comment by ehelms** [comment]
Ahh yea, I also figured out this was missing -- https://github.com/theforeman/tfm-ror51-packaging/pull/45

I updated this PR to add the tags. However, I still get when diff_package_skip is true:

```
fatal: [tfm-ror51-rubygem-uglifier]: FAILED! => {
    "failed": true
}

MSG:

The conditional check 'diff_package_changed' failed. The error was: error while evaluating conditional (diff_package_changed): 'diff_package_changed' is undefined

The error appears to have been in '/home/ehelms/workspace/obal/obal/data/roles/build_package/tasks/copr.yml': line 2, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

---
- name: 'Set test build'
  ^ here
```

**Comment by ehelms** [comment]
I got the above running:

```
 ../obal/obal/__init__.py release tfm-ror51-rubygem-uglifier --tags wait -e "diff_package_skip=true"
```

File: obal/data/roles/build_package/tasks/wait.yml [hunk 2]
```diff
@@ -1,18 +1,52 @@
...
-  when: (build_package_koji_status | failed)
+    - name: 'Failed build'
+      fail:
+        msg: "The build in {{ build_package_koji_command }}  has failed"
+      when: (build_package_koji_status | failed)
+      tags:
+        - always
+  when: build_package_type == 'koji'
+
+- block:
```
Comment (ehelms) [inline obal/data/roles/build_package/tasks/wait.yml]: Note I added this block for copr waits

**Comment by ehelms** [comment]
Full output:

```
 [WARNING]: Found both group and host with same name: localhost


PLAY [packages] ***************************************************************************************************************************************************************************************************

TASK [git_annex_setup : call git annex init] **********************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:09 -0500 (0:00:00.071)       0:00:00.071 ******** 
ok: [tfm-ror51-rubygem-uglifier]

TASK [ensure_package : check for package dir] *********************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:09 -0500 (0:00:00.247)       0:00:00.318 ******** 
ok: [tfm-ror51-rubygem-uglifier]

TASK [setup_sources : Set package_dir] ****************************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:09 -0500 (0:00:00.242)       0:00:00.561 ******** 
ok: [tfm-ror51-rubygem-uglifier]

TASK [setup_sources : Setup sources from specfile] ****************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:09 -0500 (0:00:00.018)       0:00:00.579 ******** 
included: /home/ehelms/workspace/obal/obal/data/roles/setup_sources/tasks/specfile.yml for tfm-ror51-rubygem-uglifier

TASK [setup_sources : get spec file] ******************************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:10 -0500 (0:00:00.029)       0:00:00.608 ******** 
ok: [tfm-ror51-rubygem-uglifier]

TASK [setup_sources : Set spec file path] *************************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:10 -0500 (0:00:00.241)       0:00:00.850 ******** 
ok: [tfm-ror51-rubygem-uglifier]

TASK [setup_sources : Extract Sources from spec] ******************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:10 -0500 (0:00:00.021)       0:00:00.871 ******** 
ok: [tfm-ror51-rubygem-uglifier]

TASK [setup_sources : include_tasks] ******************************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:10 -0500 (0:00:00.187)       0:00:01.059 ******** 
included: /home/ehelms/workspace/obal/obal/data/roles/setup_sources/tasks/annex.yml for tfm-ror51-rubygem-uglifier

TASK [setup_sources : Set source values] **************************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:10 -0500 (0:00:00.032)       0:00:01.091 ******** 
ok: [tfm-ror51-rubygem-uglifier]

TASK [setup_sources : Set source path] ****************************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:10 -0500 (0:00:00.022)       0:00:01.113 ******** 
ok: [tfm-ror51-rubygem-uglifier]

TASK [setup_sources : Source exists] ******************************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:10 -0500 (0:00:00.020)       0:00:01.134 ******** 
ok: [tfm-ror51-rubygem-uglifier]

TASK [setup_sources : Search annex for web link] ******************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:10 -0500 (0:00:00.141)       0:00:01.275 ******** 
ok: [tfm-ror51-rubygem-uglifier]

TASK [setup_sources : Add url to annex for source] ****************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:11 -0500 (0:00:00.472)       0:00:01.748 ******** 
skipping: [tfm-ror51-rubygem-uglifier]

TASK [setup_sources : Clone git repository] ***********************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:11 -0500 (0:00:00.019)       0:00:01.767 ******** 
skipping: [tfm-ror51-rubygem-uglifier]

TASK [diff_package : Include copr package diff] *******************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:11 -0500 (0:00:00.018)       0:00:01.786 ******** 
skipping: [tfm-ror51-rubygem-uglifier]

TASK [build_package : Include copr.yml] ***************************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:11 -0500 (0:00:00.017)       0:00:01.803 ******** 
included: /home/ehelms/workspace/obal/obal/data/roles/build_package/tasks/copr.yml for tfm-ror51-rubygem-uglifier

TASK [build_package : Set test build] *****************************************************************************************************************************************************************************
Thursday 01 March 2018  10:50:11 -0500 (0:00:00.038)       0:00:01.842 ******** 
fatal: [tfm-ror51-rubygem-uglifier]: FAILED! => {
    "failed": true
}

MSG:

The conditional check 'diff_package_changed' failed. The error was: error while evaluating conditional (diff_package_changed): 'diff_package_changed' is undefined

The error appears to have been in '/home/ehelms/workspace/obal/obal/data/roles/build_package/tasks/copr.yml': line 2, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

---
- name: 'Set test build'
  ^ here



PLAY RECAP ********************************************************************************************************************************************************************************************************
tfm-ror51-rubygem-uglifier : ok=13   changed=0    unreachable=0    failed=1  
```

**Pull request merged**